stellargraph/utils/ensemble.py

```diff
@@ -322,89 +265,41 @@ def fit_generator(
-                        validation_data=val_gen,
-                        validation_steps=validation_steps,
-                        class_weight=class_weight,
-                        max_queue_size=max_queue_size,
-                        workers=workers,
-                        use_multiprocessing=use_multiprocessing,
-                        shuffle=shuffle,
-                        initial_epoch=initial_epoch,
-                    )
+        es_callback = None
+        if use_early_stopping and validation_data is not None:
+            es_callback = [
+                EarlyStopping(
+                    monitor=early_stopping_monitor,
+                    patience=self.early_stoppping_patience,
+                    restore_best_weights=True,
                 )
-        else:
-            es_callback = None
-            if use_early_stopping and validation_generator is not None:
-                es_callback = [
-                    EarlyStopping(
-                        monitor=early_stopping_monitor,
-                        patience=self.early_stoppping_patience,
-                        restore_best_weights=True,
-                    )
-                ]
+            ]
 
-            for model in self.models:
-                self.history.append(
-                    model.fit_generator(
-                        generator=generator,
-                        steps_per_epoch=steps_per_epoch,
-                        epochs=epochs,
-                        verbose=verbose,
-                        callbacks=es_callback,
-                        validation_data=validation_generator,
-                        validation_steps=validation_steps,
-                        class_weight=class_weight,
-                        max_queue_size=max_queue_size,
-                        workers=workers,
-                        use_multiprocessing=use_multiprocessing,
-                        shuffle=shuffle,
-                        initial_epoch=initial_epoch,
-                    )
+        for model in self.models:
+            self.history.append(
+                model.fit_generator(
+                    generator=generator,
+                    steps_per_epoch=steps_per_epoch,
+                    epochs=epochs,
+                    verbose=verbose,
+                    callbacks=es_callback,
+                    validation_data=validation_data,
+                    validation_steps=validation_steps,
+                    class_weight=class_weight,
+                    max_queue_size=max_queue_size,
+                    workers=workers,
+                    use_multiprocessing=use_multiprocessing,
+                    shuffle=shuffle,
+                    initial_epoch=initial_epoch,
                 )
+            )
 
         return self.history
 
     def evaluate_generator(
         self,
         generator,
-        test_data=None,
+        test_data=None,  # Should I still allow test_data and test_targets?
```

Comment:
This comment looks more like it's asking for discussion rather than something to put here?

Reply:
Was really asking my self to give this some thought. :)